ci/build_wheel.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+package_name="cucim"
+package_dir="python/cucim"
+
+CMAKE_BUILD_TYPE="release"
+
+source rapids-configure-sccache
+source rapids-date-string
+
+# Use gha-tools rapids-pip-wheel-version to generate wheel version then
+# update the necessary files
+version_override="$(rapids-pip-wheel-version ${RAPIDS_DATE_STRING})"
+
+RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
+
+ci/release/apply_wheel_modifications.sh ${version_override} "-${RAPIDS_PY_CUDA_SUFFIX}"
+echo "The package name and/or version was modified in the package source. The git diff is:"
+git diff
+
+pip install --upgrade pip
+
+#CMake version in the container is too old, install a new version in the python venv
+pip install "cmake>=3.26.4" ninja
```

Comment:
Let's manage dependencies with `dependencies.yaml`, and list these as a build system requirement in `pyproject.toml`. See how cudf does this here:

This code...
https://github.com/rapidsai/cudf/blob/abc0d41d1d9033d581948ae19384e0aa0f33da77/dependencies.yaml#L173-L176

...generates this code:
https://github.com/rapidsai/cudf/blob/abc0d41d1d9033d581948ae19384e0aa0f33da77/python/cudf/pyproject.toml#L3-L16

Reply:
I installed it here because we are calling CMake manually via the `run build_local` commands below. `pyproject.toml` isn't being used until later for the `pip wheel` call, but we need the CMake update for the earlier `run` script steps to succeed.

We could look at refactoring to use scikit-build, but I'm not sure how much work it would be.

Reply:
Okay. Instead, use rapids-dependency-file-generator to create a requirements.txt file containing the build dependencies and install from that. We just want a single source for version information.

Reply:
Eventually I think migrating to scikit-build would be good for alignment with the rest of RAPIDS.

Reply:
This is tracked in issue: https://github.com/rapidsai/cucim/issues/305

Reply:
I think you want to do this:
```suggestion
# Install pip build dependencies (not yet using pyproject.toml)
rapids-dependency-file-generator \
  --file_key "py_build" \
  --output "requirements" \
  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee build_requirements.txt
pip install -r build_requirements.txt
```

Reply:
Thanks, it looks good!